src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -544,25 +540,27 @@ public void createServiceKey(String serviceInstanceName, String serviceKeyName,
 
     private void doCreateServiceKey(String name, Map<String, Object> parameters, Metadata metadata, CloudServiceInstance serviceInstance) {
         if (serviceInstance.getType() != ServiceInstanceType.MANAGED) {
-            throw new IllegalArgumentException("Service keys can't be created for user-provided service instances");
+            throw new IllegalArgumentException("Service keys can't be created for user-provided service instance \"" + serviceInstance.getName() + "\"");
         }
         UUID serviceInstanceGuid = getGuid(serviceInstance);
-        
+
+        var createBindingRequest = CreateServiceBindingRequest.builder()
+                                                              .type(ServiceBindingType.KEY)
+                                                              .name(name)
+                                                              .metadata(metadata)
+                                                              .relationships(ServiceBindingRelationships.builder()
+                                                                                                        .serviceInstance(buildToOneRelationship(serviceInstanceGuid))
+                                                                                                        .build());
+        if (parameters != null && !parameters.isEmpty()) {
+            createBindingRequest.parameters(parameters);
```

Comment:
Because without the condition, some tests were failing. And this preserves consistency - if no params are provided, do not pass a null/empty map to the request.